src/main/webapp/src/app/recommendation/recommendation-form/recommendation-form.component.ts

```diff
@@ -22,26 +23,72 @@ export class RecommendationFormComponent implements OnInit {
     type: ["", Validators.required]
   });
 
-  constructor(private readonly formBuilder: FormBuilder, private hiddenGemService: HiddenGemService,
-  private locationService: LocationService) { }
+  constructor(private readonly formBuilder: FormBuilder, private appComponent: AppComponent) { }
 
-  ngOnInit() {
+  ngDoCheck(){
+    if(this.hiddenGems !== this.appComponent.hiddenGems){
+      this.hiddenGems = this.appComponent.hiddenGems;
+    }
+  }
 
-    this.locationService.getLocation
-      .subscribe(location => {
-        this.location = location;
+  // Submit user's preference form
+  // Send hidden gem recommendation as output
+  async onSubmit() {
+    // filter hidden gems by preference form criteria.
+    let result = this.filterGems(this.hiddenGems, this.preferenceForm);
 
-    })
+    if(Object.keys(result).length == 0) {
+      this.errorMessage.nativeElement.innerText = "Sorry there are no hidden gems matching your preferences"
+    }
 
+    // send top 3 recommendation gems to display in list.
+    this.hiddenGemRecommendation.emit(result);
   }
-  // Submit user's preference form
-  // Send hidden gem recommendation as output
-  // TODO: function will need to be updated after findHiddenGemRecommendation will be deleted.
-  onSubmit(): void {
-    this.hiddenGemService.findHiddenGemRecommendation(this.preferenceForm.value)
-      .subscribe(hiddenGems => {
-        this.hiddenGemRecommendation.emit(hiddenGems);
-      })
+
+  filterGems(hiddenGems: HiddenGem[], preferenceForm: FormGroup): HiddenGem[] {
+
+    let recommendationGems: HiddenGem[] = hiddenGems.slice(0);
+
+    for (let i = 0; i < recommendationGems.length; i++) {
+      let gem = recommendationGems[i];
+      recommendationGems[i].matchScore = 0;
+      if(preferenceForm.controls["price"].value == recommendationGems[i].priceLevel){
```

Comment:
Do you want an exact match for price level or a <= ?

Reply:
Yes, I want exact matches to be counted. I added in an "any" option that allows all price levels (including hidden gems that have a null value)?